arrayLibClass.js

```diff
@@ -0,0 +1,239 @@
+function arrayLib() {
+
+	var passedArray;
+
+	var returnedObg = this || {};
+
+	returnedObg.chain = function (array) {
+
+		if (!arrayCheck(array)) {
+			return;
+		}
+
+
+		passedArray = array;
+		return returnedObg;
+	};
+
+	returnedObg.take = function () {
+		var array, n;
+
+		var valuesObject = argumentsCheck(arguments);
+		array = valuesObject.array;
+		n = valuesObject.serviceVar;
+
+		if (!arrayCheck(array) || !lengthCheck(array, n)) {
+			console.log('error');
+			return;
+		}
+
+		var newArr = [];
+		for (var i = 0; i < n; i++) {
+			newArr.push(array[i]);
+		}
+		passedArray = newArr;
+
+		return (arguments.length == 1) ? returnedObg : newArr;
+	};
+
+	returnedObg.skip = function () {
+		var array, n;
+
+		var valuesObject = argumentsCheck(arguments);
+		array = valuesObject.array;
+		n = valuesObject.serviceVar;
```

Comment:
try not to use such syntaxes. here you can declare variable and apply a value in the same line (check other places too)